storage/innobase/xtrabackup/test/t/innodb_keyring_file_component.sh

```diff
@@ -0,0 +1,61 @@
+#
+# Component keyring file specific variables
+#
+require_server_version_higher_than 8.0.23
+
+KEYRING_TYPE="component"
+. inc/keyring_common.sh
+. inc/keyring_file.sh
+configure_server_with_component
+
+run_cmd $MYSQL $MYSQL_ARGS test <<EOF
+CREATE TABLE t1 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t1 (c1) VALUES ('ONE'), ('TWO'), ('THREE');
+INSERT INTO t1 (c1) VALUES ('10'), ('20'), ('30');
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+ALTER INSTANCE ROTATE INNODB MASTER KEY;
+CREATE TABLE t2 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t2 SELECT * FROM t1;
+EOF
+
+xtrabackup --backup --target-dir=$topdir/backup1
+cp -r $topdir/backup1 $topdir/backup2
+record_db_state test
+shutdown_server
+
+xtrabackup --stats --datadir=${mysql_datadir} \
+  --component-keyring-file-config=${MYSQLD_DATADIR}/component_keyring_file.cnf \
+  --xtrabackup-plugin-dir=${plugin_dir}
+
+if grep -q "Encryption can't find master key, please check the keyring is loaded." $OUTFILE
+then
+  die "Cannot read component keyring file config to run stats on live server"
+fi
+
+# Test 1: Try to prepare with rubish on component config on target-dir
+cp $topdir/backup1/xtrabackup_component_keyring_file.cnf $topdir
+echo "bla" > $topdir/backup1/xtrabackup_component_keyring_file.cnf
+run_cmd_expect_failure $XB_BIN $XB_ARGS \
+--prepare --target-dir=$topdir/backup1 --xtrabackup-plugin-dir=${plugin_dir}
+
+# Test 2: Try to prepare with good component config on target-dir but
+# pass rubish as parameter
+run_cmd_expect_failure $XB_BIN $XB_ARGS \
+--prepare --target-dir=$topdir/backup2 \
+--component-keyring-file-config=$topdir/backup1/xtrabackup_component_keyring_file.cnf \
+--xtrabackup-plugin-dir=${plugin_dir}
```

Comment:
Can you paste the output of PXB command? Would like to see the error msg printed

Reply:
```
2021-05-21 09:18:47: run.sh: ===> /work/pxb/ins/8.0/xtrabackup-test/test/../../bin/xtrabackup --defaults-file=/work/pxb/ins/8.0/xtrabackup-test/test/var/w1/var1/my.cnf --no-version-check --prepare --target-dir=/work/pxb/ins/8.0/xtrabackup-test/test/var/w1/var1/backup1 --xtrabackup-plugin-dir=/work/pxb/ins/8.0/xtrabackup-test/test/../../lib/plugin/
xtrabackup: recognized server arguments: --innodb_checksum_algorithm=crc32 --innodb_log_checksums=1 --innodb_data_file_path=ibdata1:12M:autoextend --innodb_log_files_in_group=2 --innodb_log_file_size=50331648 --innodb_page_size=16384 --innodb_undo_directory=./ --innodb_undo_tablespaces=2 --server-id=1 --innodb_log_checksums=ON --innodb_redo_log_encrypt=1 --innodb_undo_log_encrypt=1
xtrabackup: recognized client arguments: --no-version-check=1 --prepare=1 --target-dir=/work/pxb/ins/8.0/xtrabackup-test/test/var/w1/var1/backup1 --xtrabackup-plugin-dir=/work/pxb/ins/8.0/xtrabackup-test/test/../../lib/plugin/
/work/pxb/ins/8.0/xtrabackup-test/test/../../bin/xtrabackup version 8.0.25-17 based on MySQL server 8.0.25 Linux (x86_64) (revision id: bc921ae98cf)
xtrabackup: cd to /work/pxb/ins/8.0/xtrabackup-test/test/var/w1/var1/backup1/
xtrabackup: This target seems to be not prepared yet.
Number of pools: 1
xtrabackup: xtrabackup_logfile detected: size=8388608, start_lsn=(18143441)
xtrabackup: using the following InnoDB configuration for recovery:
xtrabackup:   innodb_data_home_dir = .
xtrabackup:   innodb_data_file_path = ibdata1:12M:autoextend
xtrabackup:   innodb_log_group_home_dir = .
xtrabackup:   innodb_log_files_in_group = 1
xtrabackup:   innodb_log_file_size = 8388608
xtrabackup: Error: failed to init keyring component
```

Reply:
Possible to add that keyring component file passed is invalid?